### PR TITLE
Add compact IMU frame (rot+acc)

### PR DESCRIPTION
**Pull request by 9il**

### CODE REVIEW

**Review by ImUrX — APPROVED** [review]

File: server/core/src/main/java/dev/slimevr/tracking/trackers/udp/UDPProtocolParser.kt [hunk 1]
```diff
@@ -127,6 +127,7 @@ class UDPProtocolParser {
 		const val PACKET_TEMPERATURE = 20
 		const val PACKET_USER_ACTION = 21
 		const val PACKET_FEATURE_FLAGS = 22
+		const val PACKET_IMU_FRAME = 23
```
Comment (TheDevMinerTV) [inline server/core/src/main/java/dev/slimevr/tracking/trackers/udp/UDPProtocolParser.kt:130]: I'd change this to `PACKET_ROTATION_AND_ACCELERATION` for consistencies sake.

File: server/core/src/main/java/dev/slimevr/tracking/trackers/udp/UDPPacket.kt [hunk 2]
```diff
@@ -329,6 +329,25 @@ class UDPPacket22FeatureFlags(
 	}
 }
 
+data class UDPPacket23IMUFrame(
+	override var rotation: Quaternion = Quaternion.IDENTITY,
+	var acceleration: Vector3 = Vector3.NULL,
+) : UDPPacket(23),
+	RotationPacket {
+	override var sensorId: Int = 0
+	override fun readData(buf: ByteBuffer) {
```
Comment (TheDevMinerTV) [inline server/core/src/main/java/dev/slimevr/tracking/trackers/udp/UDPPacket.kt:338]:
```suggestion
	override fun readData(buf: ByteBuffer) {
		// s16 s16 s16 s16 s16 s16 s16
		// qX  qY  qZ  qW  aX  aY  aZ
```

**Review by TheDevMinerTV — APPROVED** [review]
The code looks fine, but having an actual implementation in the official firmware would be amazing.

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]
wrong button

**Comment by 9il** [comment]
@TheDevMinerTV I have renamed the packet as you suggested.

| The code looks fine, but having an actual implementation in the official firmware would be amazing.

Unfortunately, our code isn't compatible with the SlimeVR firmware due to its reliance on a special slime extension of Mir Ion, a Dlang serialization library.

However, it will be encoding will be very similar to HID after #1000 . BTW, I think the quaternion order in the HID protocol is different. Will update #1000 then.

**Review by TheDevMinerTV — APPROVED** [review]
LGTM

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**